src/test/java/net/imglib2/roi/geom/integer/LineTest.java

```diff
@@ -0,0 +1,166 @@
+package net.imglib2.roi.geom.integer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import net.imglib2.Cursor;
+import net.imglib2.Point;
+import net.imglib2.RandomAccess;
+import net.imglib2.img.Img;
+import net.imglib2.img.ImgFactory;
+import net.imglib2.img.array.ArrayImg;
+import net.imglib2.img.array.ArrayImgFactory;
+import net.imglib2.img.array.ArrayImgs;
+import net.imglib2.img.basictypeaccess.array.ByteArray;
+import net.imglib2.roi.Regions;
+import net.imglib2.type.numeric.integer.UnsignedByteType;
+import net.imglib2.util.Util;
+
+public class LineTest
```

Comment:
It would be nice to also test vertical and horizontal lines. It looks like your code should cover these edge cases, but it would be good to formally test them as well.